src/llama_server_context.cc

```diff
@@ -1546,10 +1566,15 @@ int LlamaServerContext::SplitMultipromptTask(TaskServer& multiprompt_task) {
 }
 
 void LlamaServerContext::ProcessTasks() {
-  std::unique_lock<std::mutex> lock(mutex_tasks);
-  while (!queue_tasks.empty()) {
+  while (true) {
+    std::unique_lock<std::mutex> l(mutex_tasks);
+    if (queue_tasks.empty()) {
+      l.unlock();
+      break;
+    }
     TaskServer task = queue_tasks.front();
     queue_tasks.erase(queue_tasks.begin());
+    l.unlock();
```

Comment:
Is this `unlock` for multiple model 
or just sharing the `queue_tasks` through many threads 🤔